src/Symfony/Bundle/WebProfilerBundle/Resources/views/Profiler/base.html.twig

```diff
@@ -15,10 +15,18 @@
     </head>
     <body>
         <script>
-            document.body.classList.add(
-                localStorage.getItem('symfony/profiler/theme') || (matchMedia('(prefers-color-scheme: dark)').matches ? 'theme-dark' : 'theme-light'),
-                localStorage.getItem('symfony/profiler/width') || 'width-normal'
-            );
+            if (null === localStorage.getItem('symfony/profiler/theme') || 'theme-auto' === localStorage.getItem('symfony/profiler/theme')) {
+                document.body.classList.add((matchMedia('(prefers-color-scheme: dark)').matches ? 'theme-dark' : 'theme-light'));
+            } else {
+                document.body.classList.add(localStorage.getItem('symfony/profiler/theme'));
+            }
+            // needed to respond dynamically to OS changes without having to refresh the page
+            window.matchMedia('(prefers-color-scheme: dark)').addListener(function (e) {
```

Comment:
or maybe the auto theme should use the `prefers-color-scheme` media query in the CSS directly ?

Reply:
That would be much more complex to do, so I'f prefer to avoid that. Thanks!

Reply:
why would it be more complex to support a `theme-auto` class ?